sentry/src/main/java/io/sentry/SentryExceptionFactory.java

```diff
@@ -86,7 +88,11 @@ public SentryExceptionFactory(final @NotNull SentryStackTraceFactory sentryStack
     final List<SentryStackFrame> frames =
         sentryStackTraceFactory.getStackFrames(throwable.getStackTrace());
     if (frames != null && !frames.isEmpty()) {
-      exception.setStacktrace(new SentryStackTrace(frames));
+      final SentryStackTrace sentryStackTrace = new SentryStackTrace(frames);
+      if (snapshot) {
+        sentryStackTrace.setSnapshot(true);
+      }
```

Comment:
The preference is to not add the flag at all if it isn't snapshot?

Otherwise:
```suggestion
      sentryStackTrace.setSnapshot(snapshot);
```

Reply:
yep, so no need for serialization if its null, we do the same for all the other booleans/nullable fields